### PR TITLE
SR2024 - Update Section - Robot API

### PR DESCRIPTION
**Pull request by 8BitJosh**
- Update the top level robot API page
- Add a page about the specifics of comp mode

Would specifically like some input on the page explaining comp mode

### CODE REVIEW

File: programming/robot_api/comp_mode.md [hunk 1]
```diff
@@ -5,4 +5,37 @@ title: Competition Mode
 
 # Competition Mode
 
-TODO
\ No newline at end of file
+
+## What is Comp Mode
+
+During a competition match a Competition Mode USB is inserted into a spare USB port on your robot (See the [kit assembly guide]({{ site.baseurl }}/tutorials/assembly) for details on the spare USB port you need to leave).
+This will instruct your robot that it is in competition mode and will enable a game timeout, disable the WiFi and update certain `robot` attributes.
```
Comment (RealOrangeOne) [inline programming/robot_api/comp_mode.md:12]: Nit: This sentence is a short version of what's explained below. Might be easier to just remove it and pass that off to the section below.

File: programming/robot_api/comp_mode.md [hunk 2]
```diff
@@ -5,4 +5,37 @@ title: Competition Mode
...
+
+
+## Effects of Comp Mode
+
+* Enabling Game timeout
+
+    When in competition mode the robot will automatically stop at the end of a match, where the duration of a match is defined in the [rules]({{ site.baseurl }}/tutorials/assembly).
```
Comment (RealOrangeOne) [inline programming/robot_api/comp_mode.md:19]:
```suggestion
    When in competition mode the robot will automatically stop at the end of a match, where the duration of a match is defined in the [rules]({{ site.baseurl }}/rules).
```

File: programming/robot_api/index.md [hunk 3]
```diff
@@ -3,126 +3,119 @@ layout: page
...
+    elif robot.mode == DEV:
+        print("This is development")
     ~~~~~
 
 usbkey
-:   The path to the USB memory stick.
-    Your code is unzipped and run from a temporary directory, therefore files you create will be lost when the kit is turned off.
-    You can use this to easily read from and write to files on the stick itself.
-    Note that the USB memory stick is mounted synchronously, so any writes to it will block until complete and may slow down your code.
+:   The path to the USB stick.
```
Comment (RealOrangeOne) [inline programming/robot_api/index.md:78]: Nit: `usbkey` is a `Path`, not a `str`. Should we reference this here:

```suggestion
:   A [`Path`](https://docs.python.org/3/library/pathlib.html#basic-use) object containing the path to the USB stick.
```

File: programming/robot_api/index.md [hunk 4]
```diff
@@ -3,126 +3,119 @@ layout: page
...
 mode
 :   Whether the robot is running in competition mode.
-    When in a competition match, this will be `RobotMode.COMP`, and at all other times this will be `RobotMode.DEV`.
+    When in a competition match, this will be `COMP`, and at all other times this will be `DEV`.
+
+    This attribute is only available after the start button is pressed and will throw an error if accessed before.
+    See the [competition mode](./comp_mode) page for more information about this attribute.
 
     ~~~~~ python
     from sr.robot3 import *
```
Comment (RealOrangeOne) [inline programming/robot_api/index.md:67]: Nit: Someone might accidentally read these as strings. If we explicitly import them, people know where they come from.

```suggestion
    from sr.robot3 import Robot, COMP, DEV
```

File: programming/robot_api/comp_mode.md (hunk 1 above)
Comment (8BitJosh) [reply programming/robot_api/comp_mode.md]: 1e11100

File: programming/robot_api/comp_mode.md (hunk 2 above)
Comment (8BitJosh) [reply programming/robot_api/comp_mode.md]: Fixed in 3 locations 1e11100

File: programming/robot_api/index.md (hunk 3 above)
Comment (8BitJosh) [reply programming/robot_api/index.md]: 1e11100

File: programming/robot_api/index.md (hunk 4 above)
Comment (8BitJosh) [reply programming/robot_api/index.md:67]: Discussed, not going to do

**Pull request merged**